stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -321,19 +322,22 @@ class cholesky_opencl : public vari {
    *
    */
   virtual void chain() {
-    auto L_adj_cpu = Eigen::MatrixXd::Zero(M_, M_).eval();
-    auto L_cpu = Eigen::MatrixXd::Zero(M_, M_).eval();
+    int packed_size = M_ * (M_ + 1) / 2;
+    std::vector<double> L_adj_cpu(packed_size);
+    std::vector<double> L_val_cpu(packed_size);
 
     size_t pos = 0;
     for (size_type j = 0; j < M_; ++j) {
       for (size_type i = j; i < M_; ++i) {
-        L_adj_cpu.coeffRef(i, j) = vari_ref_L_[pos]->adj_;
-        L_cpu.coeffRef(i, j) = vari_ref_L_[pos]->val_;
+        L_adj_cpu[pos] = vari_ref_L_[pos]->adj_;
+        L_val_cpu[pos] = vari_ref_L_[pos]->val_;
```

Comment:
Think we can remove the double for loop here and just iterate over the `packed_size`?

Reply:
Yeah, definitely. Fixed.